python/plugins/processing/tools/general.py

```diff
@@ -154,4 +154,5 @@ def execAlgorithmDialog(algOrName, parameters={}):
         canvas.setMapTool(prevMapTool)
 
     results = dlg.results()
+    dlg.close()
```

Comment:
Since this is not intuitively understandable for a code-reader, I'd add a comment like "make sure the dialog is destroyed and not only hidden on pressing Esc". And/or use `dlg.deleteLater()`.